ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -1061,3 +1092,29 @@ static VALUE rescued_sample_allocation(VALUE tracepoint_data) {
   // Return a dummy VALUE because we're called from rb_rescue2 which requires it
   return Qnil;
 }
+
+static void handle_allocation_sampler_error(struct cpu_and_wall_time_worker_state *state, const char *error, bool can_raise) {
+  if (error == NULL) {
+    return;
+  }
+
+  // Lets make sure our error messages explicilty mention this is for the allocation sampler. We may end up having more than one
+  // discrete sampler in the future.
+  VALUE error_rbstr = rb_str_new_cstr("allocation sampler - ");
+  rb_str_cat_cstr(error_rbstr, error);
+
+  if (can_raise) {
+    rb_raise(rb_eRuntimeError, "%"PRIsVALUE, error_rbstr);
+  } else {
+    // If we can't raise an immediate exception at the calling site, use the asynchronous flow through the main worker loop.
+    stop_state(state, rb_exc_new_str(rb_eRuntimeError, error_rbstr));
+  }
```

Comment:
Minor: Very tiny cleanup ;)

```suggestion
  VALUE exception = rb_exc_new_str(rb_eRuntimeError, rb_sprintf("allocation sampler - %s", error));

  if (can_raise) {
    rb_exc_raise(exception);
  } else {
    // If we can't raise an immediate exception at the calling site, use the asynchronous flow through the main worker loop.
    stop_state(state, exception);
  }
```